PixiEditor/Models/DataHolders/Document.cs

```diff
@@ -425,6 +426,74 @@ public void CenterContent()
                     "Center content"));
         }
 
+        /// <summary>
+        /// Merges two layers.
+        /// </summary>
+        /// <param name="firstLayer">The index of the first.</param>
+        /// <param name="secondLayer">The index of the second.</param>
+        /// <returns>The merged layer.</returns>
+        public Layer MergeLayers(Layer firstLayer, Layer secondLayer, bool nameOfSecond, int index)
+        {
+            string name;
+
+            // Wich name should be user
+            if (nameOfSecond)
+            {
+                name = secondLayer.Name;
+            }
+            else
+            {
+                name = firstLayer.Name;
+            }
+
+            Layer mergedLayer = firstLayer.MergeWith(secondLayer, name, Width, Height);
+
+            // Insert new layer and remove old
+            Layers.Insert(index, mergedLayer);
+            Layers.Remove(firstLayer);
+            Layers.Remove(secondLayer);
+
+            SetActiveLayer(Layers.IndexOf(mergedLayer));
+
+            return mergedLayer;
+        }
+
+        /// <summary>
+        /// Merges two layers.
+        /// </summary>
+        /// <param name="firstIndex">The index of the first.</param>
+        /// <param name="secondIndex">The index of the second.</param>
+        /// <returns>The merged layer.</returns>
+        public Layer MergeLayers(int firstIndex, int secondIndex, bool nameOfSecond)
+        {
+            Layer firstLayer = Layers[firstIndex];
+            Layer secondLayer = Layers[secondIndex];
+
+            return MergeLayers(firstLayer, secondLayer, nameOfSecond, firstIndex);
+        }
+
+        public Layer MergeLayers(int index, params Layer[] layers)
+        {
+            Layer finalLayer = layers[0];
+
+            for (int i = 1; i < layers.Length; i++)
+            {
+                finalLayer = finalLayer.MergeWith(layers[i], finalLayer.Name, Width, Height);
+            }
+
+            // Insert new layer and remove old
+            Layers.Insert(index, finalLayer);
+
+            foreach (Layer layer in layers)
+            {
+                Layers.Remove(layer);
+            }
+
+            SetActiveLayer(Layers.IndexOf(finalLayer));
+
+            return finalLayer;
+        }
+

```

Comment:
`Document.cs` is getting too long, maybe split some methods as other partial one or extension to `Document` class